src/Data/Conduit/Tar.hs

```diff
@@ -203,6 +210,41 @@ withEntry inner = do
             Just (ChunkException e) -> throwM e
             Nothing -> return ()
 
+{-| This function handles each entry of the tar archive according to the
+behaviour of the function passed as first argument.
+
+Here is a full example function, that read a compressed tar archive and for each entry that is a simple file, it returns its file path and SHA256 digest. Note that function can throw exceptions!
+
+> import qualified Crypto.Hash.Conduit as CH
+> import qualified Data.Conduit.Tar    as CT
+> 
+> import Conduit
+> import Crypto.Hash (Digest, SHA256)
+> import Control.Monad (when)
+> import Data.Conduit.Zlib (ungzip)
+> import Data.ByteString (ByteString)
+> 
+> filedigests :: FilePath -> IO [(FilePath, Digest SHA256)]
+> filedigests fp = runConduitRes (  sourceFileBS fp          -- read the raw file
+>                                .| ungzip                   -- gunzip
+>                                .| CT.untar                 -- decode the tar archive
+>                                .| CT.withEntries hashentry -- process each file
+>                                .| sinkList                 -- collect the results
```

Comment:
Perhaps worth instead using `mapM_C print` or `printC` to avoid holding all of the filepaths and hashes in memory.